File: bildunya_frontend/lib/data/models/user_gamification_dto.dart
import 'badge_dto.dart';

/// [com.bildunya.dto.UserGamificationDto] — Jackson `@JsonProperty` alanları.
class UserGamificationDto {
  const UserGamificationDto({
    required this.badges,
    this.currentGoalTitle,
    this.currentProgress = 0,
    this.currentGoalDetail,
    this.totalPosts = 0,
    this.distinctLocationCount = 0,
    this.verifiedContentCount = 0,
    this.mediaPostCount = 0,
    this.historicThemedCount = 0,
  });

  final List<BadgeDto> badges;
  final String? currentGoalTitle;
  final double currentProgress;
  final String? currentGoalDetail;
  final int totalPosts;
  final int distinctLocationCount;
  final int verifiedContentCount;
  final int mediaPostCount;
  final int historicThemedCount;

  factory UserGamificationDto.fromJson(Map<String, dynamic> json) {
    final rawBadges = json['badges'];
    final badges = <BadgeDto>[];
    if (rawBadges is List) {
      for (final e in rawBadges) {
        if (e is Map<String, dynamic>) {
          badges.add(BadgeDto.fromJson(e));
        }
      }
    }
    return UserGamificationDto(
      badges: badges,
      currentGoalTitle: json['current_goal_title'] as String? ??
          json['currentGoalTitle'] as String?,
      currentProgress: (json['current_progress'] as num?)?.toDouble() ??
          (json['currentProgress'] as num?)?.toDouble() ??
          0,
      currentGoalDetail: json['current_goal_detail'] as String? ??
          json['currentGoalDetail'] as String?,
      totalPosts: (json['total_posts'] as num?)?.toInt() ??
          (json['totalPosts'] as num?)?.toInt() ??
          0,
      distinctLocationCount:
          (json['distinct_location_count'] as num?)?.toInt() ??
              (json['distinctLocationCount'] as num?)?.toInt() ??
              0,
      verifiedContentCount:
          (json['verified_content_count'] as num?)?.toInt() ??
              (json['verifiedContentCount'] as num?)?.toInt() ??
              0,
      mediaPostCount: (json['media_post_count'] as num?)?.toInt() ??
          (json['mediaPostCount'] as num?)?.toInt() ??
          0,
      historicThemedCount:
          (json['historic_themed_count'] as num?)?.toInt() ??
              (json['historicThemedCount'] as num?)?.toInt() ??
              0,
    );
  }
}

File: BilDunya/src/main/java/com/bildunya/dto/UserGamificationDto.java
package com.bildunya.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserGamificationDto {

    private List<BadgeDto> badges;

    @JsonProperty("current_goal_title")
    private String currentGoalTitle;

    /** 0.0 – 1.0 */
    @JsonProperty("current_progress")
    private double currentProgress;

    @JsonProperty("current_goal_detail")
    private String currentGoalDetail;

    @JsonProperty("total_posts")
    private long totalPosts;

    @JsonProperty("distinct_location_count")
    private long distinctLocationCount;

    @JsonProperty("verified_content_count")
    private long verifiedContentCount;

    @JsonProperty("media_post_count")
    private long mediaPostCount;

    @JsonProperty("historic_themed_count")
    private long historicThemedCount;
}
